Cyber arms now spawn with the hand attached by default.

StarlightHost opened the pull request:
<!-- IT'S NOT WIZDENS REPO, IF YOU WANT TO ADD YOUR CHANGES ON ALL SERVERS, CREATE PR TO WIZDENS REPO -->

## Short description
<!-- What do you propose to change with your PR? -->

There’s also a bit of refactoring underway aimed at mechanical surgery support for IPCs.
## Why we need to add this
<!-- What is the reason for adding these changes? Please post links to Discussions as well as Bug Reports here. Please describe how this will change the game balance. -->

## Media (Video/Screenshots)
<!--
If your PR contains in-game changes you must provide screenshots/videos of the changes.
-->

## Checks
<!-- check boxes for faster reviewing of your PR -->

- [x] I do not require assistance to complete the PR.
- [x] Before posting/requesting review of a PR, I have verified that the changes work.
- [ ] I have added screenshots/videos of the changes, or this PR does not change in-game mechanics.
- [x] I affirm that my changes are licensed under the [Starlight Fork License](https://github.com/ss14Starlight/space-station-14/blob/Starlight/LICENSE-Starlight.TXT) and grant permission for use in this repository under its conditions.

**Changelog**

:cl: STARLIGHT TEAM
- add: Cyber arms now spawn with the hand attached by default.
- add: Added a button that lets admins replace their right arm with a Reaper arm, for easier surgical work.

StarlightHost committed:
Cyber arms now spawn with the hand attached by default.

Happyrobot33 commented:
this is alot, in a system I really know little about with to begin with. if everything seems to work and compile fine I'm okay merging this. Wish I could do a more proper review but I've barely touched surgery as it is, and have 0 clue the component makeup situation yet

StarlightHost commented:
@Happyrobot33 
Alright, let me break down what’s going on here:

Most of the changes are me pulling the GetSingleton method closer into our system, since I’ll be using it for some upcoming features.

There’s also a new admin Verb that force-replaces a hand with a Reaper hand, purely for testing surgery more easily, since the Reaper hand gives you a full toolset.

I added a new component: WithAttachedBodyPartsComponent. When applied to a cyber arm or leg, it automatically spawns the appropriate child part (like a hand or foot) and attaches it in the correct slot when the limb is added.

This simplifies the process. Before, you had to print an arm, print a hand, attach the arm, and then attach the hand. Now it’s all bundled.

Lastly, I updated the visual system so that when a limb has a child limb attached (like a hand on an arm), the body visual properly updates to reflect it.

Happyrobot33 commented:
> I added a new component: WithAttachedBodyPartsComponent. When applied to a cyber arm or leg, it automatically spawns the appropriate child part (like a hand or foot) and attaches it in the correct slot when the limb is added.
> 
> This simplifies the process. Before, you had to print an arm, print a hand, attach the arm, and then attach the hand. Now it’s all bundled.

conceptually I'm good with it then. my only thing would be, is it clear that if you buy the arm it'll have the hand? don't want to waste peoples money

StarlightHost committed:
review fix
